product_import_ubl/wizard/product_import.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2022 Camptocamp
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from lxml import etree
+
+from odoo import api, models
+
+CATALOGUE_NS = "urn:oasis:names:specification:ubl:schema:xsd:Catalogue-2"
+CATALOGUE_TAG = "{%s}Catalogue" % CATALOGUE_NS
+
+
+class XPathGetter(object):
+    _missing = etree.Element("Missing")
+
+    def __init__(self, element, namespaces):
+        self._xpath = element.xpath
+        self._ns = namespaces
+
+    def xpath(self, path):
+        # Return a list of elements
+        items = self._xpath(path, namespaces=self._ns)
+        return items
+
+    def xpath_get(self, path):
+        # Return 1 element
+        items = self._xpath(path, namespaces=self._ns)
+        return items[0] if items else self._missing
+
+    def xpath_text(self, path):
+        # Return text or False
+        items = self._xpath(path, namespaces=self._ns)
+        return items and items[0].text or False
+
+    get = xpath_get
+    text = xpath_text
+
+
+class ProductImport(models.TransientModel):
+    _name = "product.import"
+    _inherit = ["product.import", "base.ubl"]
```

Comment:
If you use methods from base.ubl that do not depend on an instance, please do not inherit here.
Use it as a tool like for bs doc import.

Reply:
OK, I copied it from `sale_import_ubl` I think. I don't know why tool is preferred over composition.  I'll change as you suggest.